packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -34,6 +35,359 @@ enum DayPeriod {
+
+  /// Fragments this piece is made of.
+  final List<_TimePickerHeaderFragment> fragments;
+
+  /// Index into the [fragments] list, pointing at the fragment that's centered
+  /// horizontally.
+  final int pivotIndex;
+
+  /// Vertical distance between this piece and the next piece.
+  ///
+  /// This property applies only when the header is laid out vertically.
+  final double bottomMargin;
+}
+
+_TimePickerHourFormat _getHourFormat(String formatString) {
+  switch(formatString) {
+    case TimeOfDayFormats.HH_dot_mm:
+    case TimeOfDayFormats.HH_colon_mm:
+    case TimeOfDayFormats.frenchCanadian:
+      return _TimePickerHourFormat.HH;
+    case TimeOfDayFormats.H_colon_mm:
+      return _TimePickerHourFormat.H;
+    default:
+      return _TimePickerHourFormat.h;
+  }
+}
+
+/// Provides time picker header configuration for the given [format] passing
+/// [context] to each widget in the configuration.
+_TimePickerHeaderFormat _buildHeaderFormat(String formatString, _TimePickerFragmentContext context) {
+  // Creates an hour fragment.
+  _TimePickerHeaderFragment hour(_TimePickerHourFormat hourFormat) => new _TimePickerHeaderFragment(
+    layoutId: _TimePickerHeaderId.hour,
+    widget: new _HourControl(fragmentContext: context, hourFormat: hourFormat),
+    leftMargin: _kPeriodGap,
+  );
+
+  // Creates a minute fragment.
+  _TimePickerHeaderFragment minute() => new _TimePickerHeaderFragment(
+    layoutId: _TimePickerHeaderId.minute,
+    widget: new _MinuteControl(fragmentContext: context),
+  );
+
+  // Creates a string fragment.
+  _TimePickerHeaderFragment string(_TimePickerHeaderId layoutId, String value) => new _TimePickerHeaderFragment(
+    layoutId: layoutId,
+    widget: new _StringFragment(
+      fragmentContext: context,
+      value: value,
+    ),
+  );
+
+  // Creates an am/pm fragment.
+  _TimePickerHeaderFragment dayPeriod() => new _TimePickerHeaderFragment(
+    layoutId: _TimePickerHeaderId.period,
+    widget: new _DayPeriodControl(fragmentContext: context),
+    leftMargin: _kPeriodGap,
+  );
+
+  // Convenience function for creating a time header format with up to four pieces.
+  _TimePickerHeaderFormat format(int centrepieceIndex, _TimePickerHeaderPiece p1, [_TimePickerHeaderPiece p2, _TimePickerHeaderPiece p3, _TimePickerHeaderPiece p4]) {
+    final List<_TimePickerHeaderPiece> pieces = <_TimePickerHeaderPiece>[p1];
+    if (p2 != null) {
+      pieces.add(p2);
+      if (p3 != null) {
+        pieces.add(p3);
+        if (p4 != null) {
+          pieces.add(p4);
+        }
+      }
+    }
+    return new _TimePickerHeaderFormat(centrepieceIndex, pieces);
+  }
+
+  // Convenience function for creating a time header piece with up to three fragments.
+  _TimePickerHeaderPiece piece({ int pivotIndex: -1, double bottomMargin: 0.0, _TimePickerHeaderFragment f1, _TimePickerHeaderFragment f2, _TimePickerHeaderFragment f3 }) {
```

Comment:
The documentation for `piece` and `format` make them sound like they're almost identical except for the number of fragments, but that doesn't appear to be the case (and the names suggest there's something else going on as well).